### PR TITLE
Add permission requirements to SecretManagerSecret documentation

### PR DESCRIPTION
**Pull request by gemmahou**
### Change description

This is a sample PR to add permission requirements to ConfigConnector documentation. It only applies to one doc for now, once approval, same pattern will be applied to all docs.

The idea is to have a new "before you begin" section in each resource doc, service owner should be able to identify the requirements and add there. Every resource should at least have a link to point to the service documentation "before you begin" or "prerequisites" section, including any pre-setup, grant permissions, enable apis, etc.

I also added the link to "best practice" and "troubleshooting", in the KCC resource overview page. So users can more easily find solutions to common questions.

Fixes b/291854102

### Tests you have done
The doc should now look like: 
overview: https://screenshot.googleplex.com/AL4qA6A5Qf4JgHb.png
attached cluster resource: https://screenshot.googleplex.com/6JbhdXH2FRt4jiy.png

- [x] Run `make ready-pr` to ensure this PR is ready for review.
- [N/A] Perform necessary E2E testing for changed resources.

### CODE REVIEW

File: README.NewResourceFromTerraform.md [hunk 1]
```diff
@@ -615,6 +615,11 @@ update to date, we need to add the new CRDs and samples to existing doc.
     `https://cloud.google.com` (i.e.
     `https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances`).
     Add an additional paragraph/table if needed.
+1.  Update the URL in "Before you begin" section, points to the "Prerequisites" or
```
Comment (maqiuyujoyce) [inline README.NewResourceFromTerraform.md:618]: This "Before you begin" section can't be found in a random existing file (as suggested in bullet point #1), so would be great that an example PR is linked here.

Comment (maqiuyujoyce) [inline README.NewResourceFromTerraform.md:618]: It's unclear to me that:

Do we _require_ resource contributors to add this section, or do we _suggest_ resource contributors add this section when the resource hit any conditions we set?

With the fact that only one resource has this section, I don't think we require resource contributors to do so.

I'd probably update the wording a bit to something like (I didn't use the md format in the text suggested below so please adjust it accordingly):

If additional setup is necessary before the newly added resource can be applied successfully, and the setup can't be done via the Config Connector resources or features, add a "Before You Begin" section. E.g.  https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/b019cb7c3e1dea745b804d9c6d8cb6f97fa67996/scripts/generate-google3-docs/resource-reference/generated/resource-docs/containerattached/containerattachedcluster.md

File: README.NewResourceFromTerraform.md [hunk 2]
```diff
@@ -615,6 +615,11 @@ update to date, we need to add the new CRDs and samples to existing doc.
     `https://cloud.google.com` (i.e.
     `https://cloud.google.com/spanner/docs/reference/rest/v1/projects.instances`).
     Add an additional paragraph/table if needed.
+1.  Update the URL in "Before you begin" section, points to the "Prerequisites" or 
+    "Before you begin" section in the service guide. Make sure that the URL
```
Comment (maqiuyujoyce) [inline README.NewResourceFromTerraform.md:619]: From our public doc (e.g. https://cloud.google.com/config-connector/docs/reference/resource-docs/container/containercluster), each word in the headers has its first letter capitalized.

I am not sure if it looks correct if we say "Before You Begin", but also I believe we want to keep the header format consistent. Could you check with Taryn about it? We can also just do "Prerequisites".

File: README.NewResourceFromTerraform.md [hunk 3]
```diff
@@ -615,6 +615,11 @@ update to date, we need to add the new CRDs and samples to existing doc.
...
+    is working by adding the partial URL to
+    `https://cloud.google.com` (i.e.
+    `https://cloud.google.com/spanner/docs/create-query-database-console#before`).
```
Comment (maqiuyujoyce) [inline README.NewResourceFromTerraform.md:622]: IIUC, this sentence implies that the resource contributors should:
1. Use partial URL in this section.
2. Test out whether the partial URL works by prepending `https://cloud.google.com`?

I like the idea of verification, and I understand that it is copied from the existing doc above, but I think it'll be more clear if:

1. We clarify what is the "URL" and what is the "partial URL". You seem to refer to the same thing here? I'll probably just call it "partial URL" from the beginning. E.g.

Replace the partial URL "/anthos/clusters/docs/multi-cloud/attached/eks/how-to/attach-cluster" with the partial URL pointing to the "Prerequisites" or "Before you begin" section in the resource guide. Make sure it works when prepending `https://cloud.google.com` to the partial URL. For example, if the partial URL is updated to `/spanner/docs/create-query-database-console#before`, verify if `https://cloud.google.com/spanner/docs/create-query-database-console#before` is a working URL.

2.  We suggest the verification as the last step of this section as there are multiple steps using the concept of "partial URL", and I think it'll be easier if we use a single step to talk about the verification.

File: scripts/generate-google3-docs/resource-reference/generated/resource-docs/containerattached/containerattachedcluster.md [hunk 4]
```diff
@@ -59,7 +57,11 @@ Note: Before using this resource, make sure you review and complete the steps in
 </tbody>
 </table>
 
-## Prerequisites
+## Before you begin
```
Comment (maqiuyujoyce) [inline scripts/generate-google3-docs/resource-reference/generated/resource-docs/containerattached/containerattachedcluster.md:60]: As I commented above, could you figure out a header that is consistent with the rest of the headers?

File: scripts/generate-google3-docs/resource-reference/generated/resource-docs/containerattached/containerattachedcluster.md [hunk 5]
```diff
@@ -59,7 +57,11 @@ Note: Before using this resource, make sure you review and complete the steps in
...
+
+Please review and complete <a href="/anthos/clusters/docs/multi-cloud/attached/eks/how-to/attach-cluster">Prerequisites</a>
```
Comment (maqiuyujoyce) [inline scripts/generate-google3-docs/resource-reference/generated/resource-docs/containerattached/containerattachedcluster.md:62]: Actually, if I'm reading from the users perspective, I do not know why I should do it, what the purpose I will achieve after doing it.

If this indicates that "you must follow the instructions in prerequisites before you can use this resource" then we should make it really clear in our doc. From b/291854102, it looks like we want to add prerequisite suggestions for a different resource, and that seems to be required prerequisites, and I'm a bit confused why ContainerAttachedCluster has this section. AFAIK, line #65-#105 below are already the precise instructions about what to do, then this newly added sentence is just optional, like a "More details can be found at [link to the doc]" suggestion.

I personally don't think this doc needs it.

File: scripts/generate-google3-docs/resource-reference/overview.md [hunk 6]
```diff
@@ -13,6 +13,10 @@ version. If you are using a different version, you can find each version's
 Custom Resource Definitions in the GitHub repository. For example, this
 [link contains CRDs for version 1.89.0](https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/v1.89.0/crds).
 
+Please review <a href="/config-connector/docs/best-practices">Best Practice for Config Connector</a> for best practices.
```
Comment (maqiuyujoyce) [inline scripts/generate-google3-docs/resource-reference/overview.md:16]: Replace `Config Connector` with `{{product_name_short}}`.

File: scripts/generate-google3-docs/resource-reference/overview.md [hunk 7]
```diff
@@ -13,6 +13,10 @@ version. If you are using a different version, you can find each version's
...
+
+Please review <a href="/config-connector/docs/troubleshooting">Troubleshooting Config Connector</a> for common issues.
```
Comment (maqiuyujoyce) [inline scripts/generate-google3-docs/resource-reference/overview.md:18]: Replace `Config Connector` with `{{product_name_short}}`.

Comment (maqiuyujoyce) [inline scripts/generate-google3-docs/resource-reference/overview.md:18]: Should they be new "Note"s?

**Review by maqiuyujoyce — CHANGES_REQUESTED** [review]

File: README.NewResourceFromTerraform.md (hunk 1 above)
Comment (gemmahou) [reply README.NewResourceFromTerraform.md]: Updated

File: README.NewResourceFromTerraform.md (hunk 2 above)
Comment (gemmahou) [reply README.NewResourceFromTerraform.md]: Updated to use "Prerequisites"

File: README.NewResourceFromTerraform.md (hunk 3 above)
Comment (gemmahou) [reply README.NewResourceFromTerraform.md]: I updated the existing sentences regarding partial URL, thanks for the suggestion and example! I do not think it's necessary to duplicate that in this newly added instruction.

File: scripts/generate-google3-docs/resource-reference/generated/resource-docs/containerattached/containerattachedcluster.md (hunk 4 above)
Comment (gemmahou) [reply scripts/generate-google3-docs/resource-reference/generated/resource-docs/containerattached/containerattachedcluster.md]: Changed back to Prerequisites

File: scripts/generate-google3-docs/resource-reference/generated/resource-docs/containerattached/containerattachedcluster.md (hunk 5 above)
Comment (gemmahou) [reply scripts/generate-google3-docs/resource-reference/generated/resource-docs/containerattached/containerattachedcluster.md]: Changed back, and added Prerequisites for resource secretmanagersecret

File: scripts/generate-google3-docs/resource-reference/templates/secretmanager_secretmanagersecret.tmpl [hunk 8]
```diff
@@ -49,6 +49,10 @@
 </tbody>
 </table>
 
+## Prerequisites
+Before you can use this resource, assign the Secret Manager Admin role (roles/secretmanager.admin) on the project, folder, or organization.
```
Comment (maqiuyujoyce) [inline scripts/generate-google3-docs/resource-reference/templates/secretmanager_secretmanagersecret.tmpl:53]: Could you add a new line in between the title and the paragrah?

Comment (maqiuyujoyce) [reply scripts/generate-google3-docs/resource-reference/templates/secretmanager_secretmanagersecret.tmpl:53]: Could you also run the mdlint tool for the .tmpl and .md files (hopefully they work?) to ensure the generated .md files pass the lint checks internally?

File: scripts/generate-google3-docs/resource-reference/overview.md [hunk 9]
```diff
@@ -13,6 +13,9 @@ version. If you are using a different version, you can find each version's
 Custom Resource Definitions in the GitHub repository. For example, this
 [link contains CRDs for version 1.89.0](https://github.com/GoogleCloudPlatform/k8s-config-connector/tree/v1.89.0/crds).
 
+Note: Please review <a href="/config-connector/docs/best-practices">Best Practice for {{product_name_short}}</a> for best practices.
```
Comment (maqiuyujoyce) [inline scripts/generate-google3-docs/resource-reference/overview.md:16]: Could you reuse the title of that page? I.e. `<a href="/config-connector/docs/best-practices">Best practices for {{product_name_short}}</a>`.

**Review by maqiuyujoyce — COMMENTED** [review]
Thank you for the changes, @gemmahou ! Just a few more small issues!

File: scripts/generate-google3-docs/resource-reference/templates/secretmanager_secretmanagersecret.tmpl (hunk 8 above)
Comment (gemmahou) [reply scripts/generate-google3-docs/resource-reference/templates/secretmanager_secretmanagersecret.tmpl]: Done

File: scripts/generate-google3-docs/resource-reference/overview.md (hunk 9 above)
Comment (gemmahou) [reply scripts/generate-google3-docs/resource-reference/overview.md]: Done

**Pull request merged**